### DISCUSSION TITLE
Using HTML links to link objects in Netbox UI

### DISCUSSION THREAD

**Discussion by cmsmith1977** [netbox-community/netbox#7186] [Ideas]
I often have multiple tabs open for Netbox (eg: a Tenant list, a Site list, and another for creating a new device).  I'd like to be able to right click on an object from the Sites tab, copy the link to that object, then paste that into the Site field in my Device entry tab instead of selecting from the pick list of sites.  The textbox would accept the link, pick out the actual object link, then store that link.  Might this be possible?

**Comment by candlerb** [comment]
Anything is possible given sufficient development effort :-)

I think that Netbox's web interface is still very web 1.0 in terms of its user interaction, by which I mean its reliance on forms / dialogs.  I can think of several areas where being able to drag things would be an improvement.  For me, the most useful would be to display all the interfaces/ports on two or more devices, and to be able to create connections by dragging between them; and especially to move one endpoint of a cable by dragging it to another port.

There might be some other shortcuts which help you though. Did you know it's possible to create a device by clicking in a rack elevation?  That will populate the site, rack, face and position for you.

You could use [Custom Links](https://netbox.readthedocs.io/en/stable/customization/custom-links/) to create an "Add Device" button on the Site page.  If you link to `/dcim/devices/add/?site=<id>` then you'll get the add device page with the site pre-populated.  Or you could do the same from Tenant.  That doesn't help you populate the other field though.

Also, this only works if you have the actual Site or Tenant in question open; it doesn't work from the list of Sites or list of Tenants.  I can see an argument for being able to use Custom Links from a listing page - for example, right click on an individual Site and get a drop-down of all the custom links.  In fact, even being able to go directly to the standard options of Clone and Edit from here would be very neat.

It should be relatively straightforward.  Here's a mock-up, maybe others can comment if this might be useful?  I guess it could be quite expensive to render the custom links statically for every table row, but perhaps it could be generated on-the-fly via Ajax instead.

![image](https://user-images.githubusercontent.com/44789/132193903-51e7fccb-8b0e-43b7-82f5-b8c4eaa4e822.png)
